Update "exporting_pcks" page to change our stance about mods and pcks (#10355)
godotengine/godot-docs: tutorials/export/exporting_pcks.rst

.. _doc_exporting_pcks:

Exporting packs and patches
===========================

> .. warning::

Reviewer: ```suggestion
.. danger::
```
Since the docs use `warning` for so many things, we should use `danger` for anything that is a legal, security, or data loss risk. See https://github.com/godotengine/godot-docs/pull/10252.

    We changed our stance about exporting mods using PCK files for security reasons.

>     **We don't recommend anymore using PCK files to add modding functionality to your game.**

Reviewer: ```suggestion
    **We no longer recommend using PCK files to add modding functionality to your game.**
```

>     Please read :ref:`doc_exporting_pcks_about_mods_using_pck_files` for more information.

Reviewer: ```suggestion
    See :ref:`doc_exporting_pcks_about_mods_using_pck_files` for more information.
```
Standard phrasing, avoids imperative "read".

The PCK files can even be an entirely different Godot project, which the
original game loads in at runtime.

.. _doc_exporting_pcks_about_mods_using_pck_files:

> 

Reviewer: ```suggestion

.. UPDATE: Recommendation may change. If we ever implement sandboxing or another solution
.. change this section.

```
Recently we started using these kind of comments to keep track of information in the docs that is inherently likely to change.

> About mods using PCK files

Reviewer: Usually headers are are either in the form "Using X for Y" or "Doing X with Y" or sometimes just "X", where X is the name of the feature. One of these suggestions is more idiomatic:
```suggestion
Using PCK files for modding
```
or
```suggestion
Using PCK files for mods
```
or
```suggestion
Modding games with PCK files
```
or
```suggestion
Supporting mods with PCK files
```
We should wait for other docs maintainers to comment before changing this. Note that the anchor and links to this section need to change, too.

--------------------------

While it's easy enough to create mods for a Godot game using PCK files, **we don't recommend doing it**.

The reason is simple: :ref:`PCK files are very powerful <doc_exporting_pcks_overview_of_pck_files>`. They can override and replace content in a Godot game.
While this seems ideal for modding purposes (and yes, it is), it comes with the downside of giving the keys to the castle to any bad actor.

While replacing shaders and models is quite benign, replacing scripts is problematic, as the GDScript scripting engine doesn't have actually

> any concept of a `sandbox <https://en.wikipedia.org/wiki/Sandbox_(computer_security)>`__. Every mod containing GDScript files have access

Reviewer: ```suggestion
any concept of a `sandbox <https://en.wikipedia.org/wiki/Sandbox_(computer_security)>`__. Every mod containing GDScript files has access
```

to the entire Godot Engine API, including making network calls and accessing the filesystem.

> We do currently encourage game developers to create their own modding system for the time being. To be really secure, any modding system

Reviewer: ```suggestion
We do currently encourage game developers to create their own modding system. To be really secure, any modding system
```
No need to specify both "currently" and "for the time being".

We *could* also mention the criteria for no longer recommending this (presumably some implementation of sandboxing?), but I think that doing so is optional.
